fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/logical/PushDownPredicateTest.java

```diff
@@ -242,6 +243,7 @@ public void pushDownPredicateIntoScanTest4() {
     }
 
     private Memo rewrite(Plan plan) {
-        return PlanRewriter.topDownRewriteMemo(plan, new ConnectContext(), new PushPredicateThroughJoin());
+        Plan normalizedPlan = PlanRewriter.topDownRewrite(plan, new ConnectContext(), new NormalizeExpressions());
```

Comment:
You could add a new interface in `PlanRewriter` like `topDownRewriteMemo(Plan plan, ConnectContext connectContext, List<List<RuleFactory>> ruleBatches)` to run apply rules in batches.

Reply:
This code snippet is just for test. Is it necessary to do that?

Reply:
Ok,  never mind. 
I was thinking to avoid copying in and copying out of `memo` multiple times.
Let's refactor this when we have more similar cases in the future.